src/Orleans/Core/Grain.cs

```diff
@@ -272,9 +275,25 @@ internal string CaptureRuntimeEnvironment()
     /// Base class for a Grain with declared persistent state.
     /// </summary>
     /// <typeparam name="TGrainState">The interface of the persistent state object</typeparam>
-    public class Grain<TGrainState> : Grain
+    public class Grain<TGrainState> : Grain, IStorage
         where TGrainState : class, IGrainState
     {
+        private IStorage stateStorageBridge;
+
+        private IStorage Storage
```

Comment:
This breaks DI and unit testing of grains. `IStorage` should be injected through the constructor (potentially be made part of the `IGrainRuntime`). Which implementation of `IStorage` to use should be decided on the top level (Silo initialization) and passed in.